src/gate_detect/README.md

```diff
@@ -0,0 +1,88 @@
+          --------
+```
+
+
+The gate detector used to process the image uses an interpolation function to estimate distance to the poles. Discrete data
+points must be collected of the pixel width of the poles of the gate at various distances. You then must use this data to 
+create an approximation function to map the pixel width of the poles to the distance from the pole.
+
+The current interpolation function is a power function: 
+
+(distance to pole) = interpolationConstant1 * (pixelWidthOfPole) ^ (interpolationConstant2)
+
+This function is accurate between 3 and 15 metres in the simulator, it may need to be updated with real world data points.
+
+The interpolation constants can be set in the roslaunch file.
+
+There are 7 other parameters that can be set to tune the gateDetector node. 
+
+    cannyLow: Can be set in the ros launch file and updated using ros dynamic reconfigure. Relates to the low threshold of the 
+              Canny Edge detector. Raising this value will decrease the amount of lines detected in the image by requiring a 
+              higher differential in pixel intensity to be considered an edge.
+
+    poleMax: Can be set in the ros launch file and updated using ros dynamic reconfigure. This number indicates the maximum pixel 
+              count that two lines that are detected in the image can be to be considered a pole of the gate. Decreasing this 
+              value will decrease the amount of poles detected.
+
+    houghLinesThreshold: Can be set in the ros launch file and updated using ros dynamic reconfigure. This number sets the 
+                          maximum required line intersections required for a line to be detected in the opencv hough line
+                          detector function. Increasing this number will decrease the amount of lines detected.
+
+    houghLinesMinLength: Can be set in the ros launch file and updated using ros dynamic reconfigure. This number indicates the
+                          minimum length a line can be to be considered a line by the opencv hough line detector.
+
+    houghLinesMaxLineGap: Can be set in the ros launch file and updated using ros dynamic reconfigure. This number indicates the
+                          maximum gap that can occur between 2 lines to be considered 1 line by the open cv hough line detector.
+
+    lowVertThresh: This number cannot be set in a ros launch file or dynamically reconfigured. It represents the maximum
+                   deviation from perfectly vertical that a line can have to be considered a vertical line. Increasing this 
+                   number will increase the amount of lines detected.
+
+    lowHorThresh: This number cannot be set in a ros launch file or dynamically reconfigured. It represents the maximum
+                   deviation from perfectly horizontal that a line can have to be considered a horizontal line. Increasing 
+                   this number will increase the amount of lines detected.
```

Comment:
This is great! I might suggest more tightly coupling the parameter descriptions here with the code itself, probably by putting them in an example roslaunch file.

Reply:
done, that's a good idea